src/compiler/checker.ts

```diff
@@ -14624,68 +14653,49 @@ namespace ts {
                     return target.kind === SyntaxKind.SuperKeyword;
                 case SyntaxKind.PropertyAccessExpression:
                 case SyntaxKind.ElementAccessExpression:
-                    return (isPropertyAccessExpression(target) || isElementAccessExpression(target)) &&
-                        getAccessedPropertyName(source as PropertyAccessExpression | ElementAccessExpression) === getAccessedPropertyName(target) &&
-                        isMatchingReference((source as PropertyAccessExpression | ElementAccessExpression).expression, target.expression);
-                case SyntaxKind.BindingElement:
-                    if (target.kind === SyntaxKind.PropertyAccessExpression && (<PropertyAccessExpression>target).name.escapedText === getBindingElementNameText(<BindingElement>source)) {
-                        const ancestor = source.parent.parent;
-                        if (ancestor.kind === SyntaxKind.BindingElement) {
-                            return isMatchingReference(ancestor, (<PropertyAccessExpression>target).expression);
-                        }
-                        if (ancestor.kind === SyntaxKind.VariableDeclaration) {
-                            const initializer = (<VariableDeclaration>ancestor).initializer;
-                            return !!initializer && isMatchingReference(initializer, (<PropertyAccessExpression>target).expression);
-                        }
-                    }
+                    return isAccessExpression(target) &&
+                        getAccessedPropertyName(<AccessExpression>source) === getAccessedPropertyName(target) &&
+                        isMatchingReference((<AccessExpression>source).expression, target.expression);
             }
             return false;
         }
 
-        function getAccessedPropertyName(access: PropertyAccessExpression | ElementAccessExpression): __String | undefined {
-            return isPropertyAccessExpression(access) ? access.name.escapedText :
+        function getAccessedPropertyName(access: AccessExpression): __String | undefined {
+            return access.kind === SyntaxKind.PropertyAccessExpression ? access.name.escapedText :
                 isStringLiteral(access.argumentExpression) || isNumericLiteral(access.argumentExpression) ? escapeLeadingUnderscores(access.argumentExpression.text) :

```

Comment:
While we're here, we should probably change this to `isStringOrNumericLiteralLike` so `NoSubstitutionTemplateLiteral`s can act like string literals.